src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -188,6 +188,21 @@ public static Set<Illness> parseIllnesses(Collection<String> illnesses) throws P
         return illnessSet;
     }
 
+    /**
+     * Parses a {@code String illness} into a {@code String}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code String} is invalid.
+     */
+    public static String parseFindIllness(String illness) throws ParseException {
+        requireNonNull(illness);
+        String trimmedIllness = illness.trim();
+        if (!Illness.isValidIllnessFind(trimmedIllness)) {
+            throw new ParseException(Illness.FIND_MESSAGE_CONSTRAINTS);
+        }
+        return trimmedIllness;
+    }
+
```

Comment:
Awesome! Seems like you have improved the parsing logic to not create a new Illness logic with an invalid IllnessName. 💯